### PR TITLE
Improve internal AQL and transaction timeouts

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Increase internal AQL query and transaction timeout on DB servers from 3 to 5 minutes.
Previously, queries and transactions on DB servers could expire quickly, which led to spurious "query ID not found" or "transaction ID not found" errors on DB servers for multi-server queries/transactions with unbalanced access patterns for the different participating DB servers.

The timeouts on coordinators remain unchanged, so any queries/transactions that are abandoned will be aborted there, which will also be propagated to DB servers. In addition, if a particpating server in an AQL query becomes unavailable, the coordinator is now notified of that and will terminate the query more eagerly.

This PR addresses the timeouts issue described in https://arangodb.atlassian.net/browse/BTS-439

This is a change with potentially large side-effects, so there are no plans to backport it to earlier versions.

This PR is still missing useful tests.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: arangod/Aql/EngineInfoContainerDBServerServerBased.cpp [hunk 1]
```diff
@@ -408,7 +427,6 @@ Result EngineInfoContainerDBServerServerBased::buildEngines(
   options.database = _query.vocbase().name();
   options.timeout = network::Timeout(SETUP_TIMEOUT);
   options.skipScheduler = true;  // hack to speed up future.get()
-  options.param("ttl", std::to_string(_query.queryOptions().ttl));
```
Comment (jsteemann) [inline arangod/Aql/EngineInfoContainerDBServerServerBased.cpp:411]: note: this request parameter was set, but not read by RestAqlHandler.

File: arangod/Aql/ExecutionEngine.cpp [hunk 2]
```diff
@@ -490,6 +494,45 @@ struct DistributedQueryInstanciator final
...
+      ClusterInfo& ci = _query.vocbase().server().getFeature<ClusterFeature>().clusterInfo();
+      engine->rebootTrackers().reserve(srvrQryId.size());
+      for (auto const& srvr : srvrQryId) {
+        std::string comment = std::string("AQL query from coordinator ") + ServerState::instance()->getId();
+        // intentional copy, because we may modify the value later
+        auto server = srvr.first;
+        if (server.compare(0, 7, "server:") == 0) {
+          server = server.substr(7);
+        }
+        auto rebootId = ci.rebootTracker().rebootId(server);
```
Comment (goedderz) [inline arangod/Aql/ExecutionEngine.cpp:515]: @jsteemann I might be missing something, but I think taking the RebootId this way is subject to a race. It must either be taken from the answer of the server when creating the snippet, or the expected RebootId (i.e. the one we're using here) must be sent with when creating the snippet for the respective DBServer to check and decline the request if it differs.

Thanks for the PR btw, this is great progress for AQL.

Comment (jsteemann) [reply arangod/Aql/ExecutionEngine.cpp]: @goedderz: I made this change in the meantime.
If you want to, please re-review. Thanks!

File: arangod/Cluster/RebootTracker.h [hunk 3]
```diff
@@ -74,11 +75,6 @@ class RebootTracker {
 
   void updateServerState(std::unordered_map<ServerID, RebootId> const& state);
 
-
-  /// @brief determine the current reboot id for a server.
-  /// will throw if no reboot id is known for the server
-  RebootId rebootId(ServerID const& server) const;
```
Comment (goedderz) [inline arangod/Cluster/RebootTracker.h]: :+1:

**Review by goedderz — APPROVED** [review]

**Pull request merged**